pallets/liquid-staking/src/benchmarking.rs

```diff
@@ -0,0 +1,251 @@
+//! Liquid Staking pallet benchmarking.
+
+#![cfg(feature = "runtime-benchmarks")]
+use super::*;
+
+use crate::Pallet as LiquidStaking;
+
+use frame_benchmarking::account;
+use frame_benchmarking::{benchmarks, impl_benchmark_test_suite, whitelisted_caller};
+use frame_support::sp_runtime::FixedPointNumber;
+use frame_support::{
+    assert_ok,
+    traits::{Get, UnfilteredDispatchable},
+};
+use frame_system::{self, RawOrigin as SystemOrigin};
+use orml_traits::MultiCurrency;
+use primitives::Rate;
+
+fn assert_last_event<T: Config>(generic_event: <T as Config>::Event) {
+    frame_system::Pallet::<T>::assert_last_event(generic_event.into());
+}
+
+const INITIAL_AMOUNT: u128 = 1000_000_000_000_000;
+const SEED: u32 = 0;
+
+fn initial_set_up<T: Config>(caller: T::AccountId) {
+    let currency_id = T::StakingCurrency::get();
+    let account_id = LiquidStaking::<T>::account_id();
+    T::Currency::deposit(currency_id, &caller, INITIAL_AMOUNT).unwrap();
+    T::Currency::deposit(currency_id, &account_id, INITIAL_AMOUNT).unwrap();
+    ExchangeRate::<T>::put(Rate::saturating_from_rational(2, 100));
+}
+
+benchmarks! {
+
+    stake {
+        let caller: T::AccountId = whitelisted_caller();
+        initial_set_up::<T>(caller.clone());
+        let amount = 100_000;
+    }: _(SystemOrigin::Signed(caller.clone()), amount)
+    verify {
+        assert_eq!(TotalStakingAsset::<T>::get(), amount);
+        assert_eq!(TotalVoucher::<T>::get(), 5_000_000);
+
+        // Check balance is correct
+        assert_eq!(
+            <T as Config>::Currency::free_balance(T::StakingCurrency::get(), &caller),
+            INITIAL_AMOUNT - amount
+        );
+        assert_eq!(
+            <T as Config>::Currency::free_balance(T::LiquidCurrency::get(), &caller),
+            5_000_000
+        );
+        assert_eq!(
+            <T as Config>::Currency::free_balance(T::StakingCurrency::get(), &LiquidStaking::<T>::account_id()),
+            INITIAL_AMOUNT + amount
+        );
+        assert_last_event::<T>(Event::<T>::Staked(caller, amount).into());
+    }
+
+    withdraw {
+        let caller: T::AccountId = whitelisted_caller();
+        initial_set_up::<T>(caller.clone());
+        let agent: T::AccountId = account("Sample", 6, SEED);
+        let amount = 100_000;
+        let withdraw_amount = T::MaxWithdrawAmount::get() - 10;
+        assert_ok!(LiquidStaking::<T>::stake(
+            SystemOrigin::Signed(caller.clone()).into(),
+            amount));
+        let call = Call::<T>::withdraw(agent.clone(), withdraw_amount);
+        let origin = SystemOrigin::Root.into();
+    }: { call.dispatch_bypass_filter(origin)? }
```

Comment:
Most of the time to call withdraw or slash, we aren't using root, but a member of council.

Reply:
As discussed, I will be using `T::WithdrawOrigin::successful_origin();`